gym/spaces/box.py

```diff
@@ -50,11 +50,30 @@ def __init__(self, low, high, shape=None, dtype=np.float32, seed=None):
                 "shape must be provided or inferred from the shapes of low or high"
             )
 
+        # return custom infinite if is integer infinite
+        def _get_inf(dtype):
+            if np.dtype(dtype).kind == "f":
+                return np.inf
+            elif np.dtype(dtype).kind == "i":
+                temp = np.ones((1,)).astype(dtype)
+                return 2 ** (temp[0].nbytes * 8 - 1) - 2
```

Comment:
This looks a bit iffy. The idea is to get the maximum/minimum integer, yes? Afaik the "right" way to do it is with `np.iinfo`